### PR TITLE
Allows to specify SMTP encryption

### PR DESCRIPTION
**Pull request by D3nnisH**

### CODE REVIEW

File: src/main/java/sirius/web/mails/SMTPConfiguration.java [hunk 1]
```diff
@@ -158,6 +168,15 @@ public String getMailPort() {
         return port;
     }
 
+    /**
+     * Returns the encryption method used to connect to the mail server
+     *
+     * @return the encryption method used to connect to the mail server
+     */
+    public SMTPEncryption getMailEncryption() {
```
Comment (andyHa) [inline src/main/java/sirius/web/mails/SMTPConfiguration.java]: nope this is the transport protocol which might be encrypted

File: src/main/java/sirius/web/mails/SMTPConfiguration.java [hunk 2]
```diff
@@ -106,6 +115,7 @@ public static SMTPConfiguration fromUserSettings() {
     public static SMTPConfiguration fromSettings(Settings settings) {
         return new SMTPConfiguration(settings.get("mail.host").getString(),
                                      settings.get("mail.port").getString(),
+                                     settings.get("mail.encryption").asEnum(SMTPEncryption.class),
```
Comment (andyHa) [inline src/main/java/sirius/web/mails/SMTPConfiguration.java]: protocol

File: src/main/java/sirius/web/mails/SMTPConfiguration.java [hunk 3]
```diff
@@ -31,6 +33,9 @@
     @ConfigValue("mail.smtp.port")
     private static String smtpPort;
 
+    @ConfigValue("mail.smtp.encryption")
```
Comment (andyHa) [inline src/main/java/sirius/web/mails/SMTPConfiguration.java]: see below..

File: src/main/java/sirius/web/mails/SMTPConfiguration.java [hunk 4]
```diff
@@ -206,4 +225,38 @@ public String getMailSenderName() {
     public boolean isUseSenderAndEnvelopeFrom() {
         return useSenderAndEnvelopeFrom;
     }
+
+    /**
+     * Lists encryption methods for SMTP.
```
Comment (andyHa) [inline src/main/java/sirius/web/mails/SMTPConfiguration.java]: protocols

File: src/main/java/sirius/web/mails/SMTPConfiguration.java [hunk 5]
```diff
@@ -206,4 +225,38 @@ public String getMailSenderName() {
...
+     */
+    public enum SMTPEncryption {
+        NONE("smtp", false),
```
Comment (andyHa) [inline src/main/java/sirius/web/mails/SMTPConfiguration.java]: add a short description per constant

File: src/main/java/sirius/web/mails/SMTPConfiguration.java [hunk 6]
```diff
@@ -206,4 +225,38 @@ public String getMailSenderName() {
...
+
+        public String getProtocol() {
+            return protocol;
+        }
+
+        public boolean isStarttls() {
+            return starttls;
+        }
+
+        public boolean isEncrypted() {
```
Comment (andyHa) [inline src/main/java/sirius/web/mails/SMTPConfiguration.java]: supports encryption

File: src/main/java/sirius/web/mails/SMTPConfiguration.java [hunk 7]
```diff
@@ -206,4 +225,38 @@ public String getMailSenderName() {
...
+            return starttls || "smtps".equals(protocol);
+        }
+
+        @Override
+        public String toString() {
+            return NLS.get(SMTPEncryption.class.getSimpleName() + "." + name());
```
[the remark this answers is not in the document]
Comment (D3nnisH) [reply src/main/java/sirius/web/mails/SMTPConfiguration.java]: no, this is the purpose of the `.name()` method

File: src/main/resources/web_de.properties [hunk 8]
```diff
@@ -48,6 +48,9 @@ RythmConfig.jsTimeFormat = H:i
 ScopeDefaultConfigController.config = Standard-Konfiguration
 ScopeDefaultConfigController.files = Datei
 SmartLineBasedProcessor.tooManyCols = Zu viele Spalten (${cols})!
+SMTPEncryption.NONE = Unverschlüsselt
```
Comment (andyHa) [inline src/main/resources/web_de.properties]: return (untranslated) SMTP, STMPS, STMP+STARTTLS

File: src/main/resources/component-web.conf [hunk 9]
```diff
@@ -339,6 +339,9 @@ mail {
         # Port used by the SMTP server
         port = "25"
 
+        # Protocol and encryption of the SMTP connection (SMTP, SMTPS or STARTTLS)
+        protocol = ""
```
Comment (andyHa) [inline src/main/resources/component-web.conf]: shouldn't this be "smtp" to be backwards compatible?

**Pull request merged**